### PR TITLE
Add functional tests build step.

### PR DESCRIPTION
**Pull request by glogiotatidis**
This is already used in https://ci.us-west.moz.works/job/bedrock_functional_tests/

/cc @pmclanahan @jgmize @davehunt 

- [ ] Update Jenkins job to use these scripts after merging.

### CODE REVIEW

File: docker/jenkins/run_functional_tests.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -xe
+
+cp docker/dockerfiles/bedrock_functional_tests Dockerfile
+docker build -t bedrock_functional_tests:$GIT_COMMIT --pull=true .
+
+docker pull selenium/hub:2.48.2
+docker pull selenium/node-firefox:2.48.2
```
Comment (davehunt) [inline docker/jenkins/run_functional_tests.sh]: We'll want to update the tag when new releases are made. Perhaps we should use an environment variable, as we use it in four places already?

File: docker/jenkins/run_functional_tests.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+docker run -d --name bedrock-code-${BUILD_NUMBER} -e ALLOWED_HOSTS="*" -e SECRET_KEY=foo -e DEBUG=False -e DATABASE_URL=sqlite:////tmp/temp.db mozorg/bedrock_code:$GIT_COMMIT
+docker run -d --name bedrock-selenium-hub-${BUILD_NUMBER} selenium/hub:2.48.2
+docker run -d --name bedrock-selenium-node-${BUILD_NUMBER} --link bedrock-selenium-hub-${BUILD_NUMBER}:hub --link bedrock-code-${BUILD_NUMBER}:bedrock selenium/node-firefox:2.48.2
```
Comment (davehunt) [inline docker/jenkins/run_functional_tests.sh]: We can probably afford to spin up a few nodes (perhaps 5?) so tests can run in parallel. We could also do this via Docker compose to start the various containers and use [scale](https://docs.docker.com/compose/reference/scale/) for the nodes.

**Comment by davehunt** [comment]
Awesome work @glogiotatidis! :smile_cat:

**Comment by jgmize** [comment]
@glogiotatidis would you mind taking a look at https://ci.us-west.moz.works/job/bedrock_functional_tests/lastFailedBuild/console?

**Comment by davehunt** [comment]
I just noticed a failure in the pipeline, and it may be caused by the tests running before the Selenium nodes are ready.

File: docker/jenkins/run_functional_tests.sh (hunk 1 above)
Comment (glogiotatidis) [inline docker/jenkins/run_functional_tests.sh]: I added an env variable, although I like to make changes in the repo when possible so we have history. We can do either way now.

File: docker/jenkins/run_functional_tests.sh (hunk 2 above)
Comment (glogiotatidis) [inline docker/jenkins/run_functional_tests.sh]: Added a loop controlled by an env variable that defaults to 5. ATM it seems that although we spin more node we still need about 75' to run the tests but I guess we'll see the benefits of more node as the tests increase

**Comment by glogiotatidis** [comment]
Added a check that wait until at least one node of the started nodes is ready.

This is ready for another look

**Comment by davehunt** [comment]
Nice. r+

**Comment by jgmize** [comment]
Nice work @glogiotatidis and thanks @davehunt!

**Pull request merged**